app/services/PageRenderer.scala

```diff
@@ -60,7 +75,11 @@ class PageRenderer @Inject() () {
     stanzaMap.get(stanzaId) match {
       case None => (visualStanzas, labels, seen, stanzaId, None)
       case Some(s) => s match {
-        case EndStanza => (visualStanzas, labels, seen :+ stanzaId, stanzaId, None)
+        case EndStanza =>
+          labels.takeFlow match {
+            case Some((nxt, stanzas, updatedLabels)) => evaluateStanzas(nxt, updatedLabels, visualStanzas, seen)
```

Comment:
This line is not currently being executed by the tests; do we need two flows with no input?

Reply:
Safety line in case this situation ocurrs.